app/scripts.babel/popup.js

```diff
@@ -267,12 +267,29 @@ limitations under the License.
 
       let onUI;
       let nuxeo;
+      let repo;
+
+      const regexes = {};
+      regexes.uuid = /[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}/;
+      regexes.repo = /[A-Za-z_\.0-9-]+/;
+      regexes.path = /\/[A-Za-z\.0-9_\- \/%~:?#'"]+/;
+
+      regexes.jsf = {};
+      regexes.jsf.nxpath = new RegExp(`(?:nxpath/(?<repoPath>${regexes.repo.source})(?<path>${regexes.path.source}))`);
+      regexes.jsf.nxdoc = new RegExp(`(?:nxdoc/(?<repoId>${regexes.repo.source})/(?<docid>${regexes.uuid.source}))`);
+      regexes.jsf.doc = new RegExp(`/nuxeo/(${regexes.jsf.nxpath.source}|${regexes.jsf.nxdoc.source})`);
+
+      regexes.ui = {};
+      regexes.ui.browse = new RegExp(`(?:browse(?<path>\/.+?(?=\\?|$)))`);
+      regexes.ui.docid = new RegExp(`(?:doc[\/A-Za-z_\.0-9]+(?<docid>${regexes.uuid.source}))`);
+      regexes.ui.doc = new RegExp(`/nuxeo(?:/repo/(?<repo>${regexes.repo.source})|)/ui/#!/(?:${regexes.ui.browse.source}|${regexes.ui.docid.source})`);
 
       function getJsonFromPath(input) {
         input = decodeURIComponent(input);
         chrome.storage.sync.get('highlight', (res) => {
           if (res.highlight) {
-            nuxeo.request(`/path${input}`)
+            const path = repo ? `/repo/${repo}/path${input}` : `/path${input}`;
```

Comment:
instead of using the ternary operator each time an URL is implied; we can assume `repo` has a default value: `default` and append it to every URL.